calabash-cucumber/lib/calabash-cucumber/store/preferences.rb

```diff
@@ -13,14 +14,80 @@ def initialize
         @path = File.join(dot_dir, "preferences", "preferences.json")
       end
 
+      # !@visibility private
+      def usage_tracking
+        preferences = read
+
+        unless valid_user_tracking_value?(preferences[:usage_tracking])
+          log_defaults_reset
+          preferences[:usage_tracking] = defaults[:usage_tracking]
+          write(preferences)
+        end
+
+        preferences[:usage_tracking]
+      end
+
+      # !@visibility private
+      def usage_tracking=(value)
+        if !valid_user_tracking_value?(value)
+          raise ArgumentError,
+            "Expected '#{value}' to be one of #{VALID_USAGE_TRACKING_VALUES.join(", ")}"
+        end
+
+        preferences = read
+        preferences[:usage_tracking] = value
+        write(preferences)
+      end
+
+      # !@visibility private
+      def user_id
+        preferences = read
+
+        unless valid_user_id?(preferences[:user_id])
+          preferences[:user_id] = SecureRandom.uuid
+          write(preferences)
+        end
+
+        preferences[:user_id]
+      end
+
+      # !@visibility private
+      def user_id=(value)
+        if !valid_user_id?(value)
+          raise ArgumentError,
+            "Expected '#{value}' to not be nil and not an empty string"
+        end
+
+        preferences = read
+        preferences[:user_id] = value
+        write(preferences)
+      end
+
       private
 
       # @!visibility private
+      def valid_user_tracking_value?(value)
+        VALID_USAGE_TRACKING_VALUES.include?(value)
```

Comment:
Might want to consider doing some normalization about `value` in case a user mistakenly puts `None` or `null` or something for the `user_tracking` key.